gdpr/anonymizers/base.py

```diff
@@ -0,0 +1,169 @@
+        value = version.field_dict[name]
+        if self._ignore_empty_values and value in self._empty_values:
+            return value
+        self._encryption_key = encryption_key
+        return self.get_anonymized_value(value)
+
+    def get_deanonymized_value_from_version(self, obj, version, name: str, encryption_key: Optional[str] = None):
+        if self.get_is_reversible(obj) is False:
+            raise self.IrreversibleAnonymizationException()
+        value = version.field_dict[name]
+        if self._ignore_empty_values and value in self._empty_values:
+            return value
+        self._encryption_key = encryption_key
+        return self.get_decrypted_value(value)
+
+    def get_encryption_key(self):
+        """Return encryption key to use.
+
+        Override this method for custom logic.
+        """
+        if self._encryption_key is None:
+            raise ImproperlyConfigured()
+        return self._encryption_key
+
+    def get_anonymized_value(self, value: Any) -> Any:
+        """
+        There must be defined implementation of rule for anonymization.
+        To retain compatibility with older anonymizers this method is used as redirect to
+        `get_encrypted_value`.
+
+        Args:
+            value: value that is anonymize
+
+        Returns:
+            anonymized value
+        """
+        return self.get_encrypted_value(value)
+
+    def get_encrypted_value(self, value: Any) -> Any:
+        """
+        There must be defined implementation of rule for anonymization
+
+        Args:
+            value: value that is anonymize
+
+        Returns:
+            anonymized value
+        """
+        raise NotImplementedError
+
+    def get_decrypted_value(self, value: Any) -> Any:
+        """
+        There must be defined implementation of rule for anonymization
+
+        Args:
+            value: value that is anonymize
+
+        Returns:
+            anonymized value
+        """
+        if self.is_reversible:
+            raise NotImplementedError
+        return self.IrreversibleAnonymizationException()
+
+
+class NumericFieldAnonymizer(FieldAnonymizer):
+    max_anonymization_range: int
+
+    def __init__(self, max_anonymization_range: int = None, ignore_empty_values: bool = None,
+                 empty_values: Optional[List[Any]] = None):
+        if max_anonymization_range:
+            self.max_anonymization_range = max_anonymization_range
+        super().__init__(ignore_empty_values, empty_values)
+
+    def get_numeric_encryption_key(self, value: Union[int, float] = None) -> int:
+        if value is None:
+            return numerize_key(self.get_encryption_key()) % self.max_anonymization_range
+        # safety measure against key getting one bigger (overflow) on decrypt e.g. (5)=1 -> 5 + 8 = 13 -> (13)=2
+        guess_len = len(str(int(value)))
+        return numerize_key(self.get_encryption_key()) % 10 ** (guess_len if guess_len % 2 != 0 else (guess_len - 1))
```

Comment:
můžeš nějak popsat to guess_len. Proč to řeší sudá/lichá? Snažíš se nějak zachovat délku čísla, ale proč je to jiné pro sudé a liché délky

Reply:
Ta metoda si zaslouží docstring dokumentaci. Co je třeba to `value`?

Reply:
@matllubos 

Jde o to jak je popsané když mám value třeba `5` a encryption_key který se k tomu přičítá by byl `8` tak `5 + 8 => 13` tudíž hodnota změnila svou délku a při rozšifrování by se použil šifrovací klíč o délce dva místo 1 a vyšla by tedy míso `encryption_key=8` třeba `encryption_key=18` a tudíž místo původních `5` bychom se dostali k `-5`